### PR TITLE
Remove stock level callsides from shipment events

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

File: core/app/models/spree/shipment.rb [hunk 1]
```diff
@@ -58,7 +57,10 @@ class Shipment < Spree::Base
         }
         transition from: :canceled, to: :pending
       end
-      after_transition from: :canceled, to: [:pending, :ready, :shipped], do: :after_resume
+
+      event :ship do
```
Comment (mbj) [inline core/app/models/spree/shipment.rb]: We need to explicitly define this transition that was implicitly defined by the `after_transition` blocks we remove in this PR.

Comment (dkubb) [inline core/app/models/spree/shipment.rb]: I assume without this change the specs/functionality breaks?

**Comment by dkubb** [comment]
@mbj I left one comment in this PR, but otherwise I see no blocker for merging.

File: core/app/models/spree/shipment.rb (hunk 1 above)
Comment (mbj) [inline core/app/models/spree/shipment.rb]: Actually, the ship event was removed on propose in https://github.com/MountainRoseHerbs/spree/pull/290. And I should only bring the required `#shipped?` back (or recurse into its call sides), not the entire event.

I'll do another cycle.

File: core/app/models/spree/order_inventory.rb [hunk 2]
```diff
@@ -71,16 +71,11 @@ def add_to_shipment(shipment, quantity)
...
-        if order.completed?
-          shipment.stock_location.unstock(variant, quantity, shipment)
-        end
-
         quantity
       end
 
       def remove_from_shipment(shipment, quantity)
-        return 0 if quantity == 0 || shipment.shipped?
+        return 0 if quantity == 0
```
Comment (dkubb) [inline core/app/models/spree/order_inventory.rb]: You could do: `return 0 if quantity.zero?` if you wanted.

Or even: `return quantity if quantity.zero?`

**Pull request closed without merging**